src/pgen/radiation_advection.cpp

```diff
@@ -39,32 +41,65 @@ void ProblemGenerator(MeshBlock *pmb, ParameterInput *pin) {
   auto iphi = imap.GetFlatIdx(radmoment_internal::phi);
   const int prho = imap[fluid_prim::density].first;
   const int pT = imap[fluid_prim::temperature].first;
+  const int peng = imap[fluid_prim::energy].first;
+
+  Params &phoebus_params = pmb->packages.Get("phoebus")->AllParams();
+
+  auto eos = pmb->packages.Get("eos")->Param<singularity::EOS>("d.EOS");
+  const auto opac_d =
+      pmb->packages.Get("opacity")->template Param<singularity::neutrinos::Opacity>(
+          "d.opacity");
+  auto &unit_conv =
+      pmb->packages.Get("phoebus")->Param<phoebus::UnitConversions>("unit_conv");
+  const Real MASS = unit_conv.GetMassCGSToCode();
+  const Real LENGTH = unit_conv.GetLengthCGSToCode();
+  const Real RHO = unit_conv.GetMassDensityCGSToCode();
+  const Real TEMP = unit_conv.GetTemperatureCGSToCode();
 
   const auto specB = idJ.GetBounds(1);
-  const Real J = pin->GetOrAddReal("radiation_advection", "J", 1.0);
+  const bool scale_free = pin->GetOrAddBoolean("units", "scale_free", true);
+  const Real J0 = pin->GetOrAddReal("radiation_advection", "J", 1.0);
   const Real Hx = pin->GetOrAddReal("radiation_advection", "Hx", 0.0);
   const Real Hy = pin->GetOrAddReal("radiation_advection", "Hy", 0.0);
   const Real Hz = pin->GetOrAddReal("radiation_advection", "Hz", 0.0);
+  PARTHENON_DEBUG_REQUIRE(std::fabs(Hx) < J0 && std::fabs(Hy) < J0 && std::fabs(Hz) < J0,
+                          "Fluxes are incorrectly normalized!");
   const Real vx = pin->GetOrAddReal("radiation_advection", "vx", 0.0);
+  PARTHENON_REQUIRE(std::fabs(vx) < 1., "Subluminal velocity required!");
   const Real width = pin->GetOrAddReal("radiation_advection", "width", sqrt(2.0));
-  const Real kappa = pin->GetOrAddReal("radiation_advection", "kappas_init", 1.e3);
+  const Real kappa = pin->GetReal("opacity", "gray_kappa") * LENGTH * LENGTH / MASS;
   const bool boost = pin->GetOrAddBoolean("radiation_advection", "boost_profile", false);
   const int shapedim = pin->GetOrAddInteger("radiation_advection", "shapedim", 1);
 
   auto &coords = pmb->coords;
   auto pmesh = pmb->pmy_mesh;
   int ndim = pmesh->ndim;
 
-  IndexRange ib = pmb->cellbounds.GetBoundsI(IndexDomain::entire);
-  IndexRange jb = pmb->cellbounds.GetBoundsJ(IndexDomain::entire);
-  IndexRange kb = pmb->cellbounds.GetBoundsK(IndexDomain::entire);
+  IndexRange ib = pmb->cellbounds.GetBoundsI(IndexDomain::interior);
+  IndexRange jb = pmb->cellbounds.GetBoundsJ(IndexDomain::interior);
+  IndexRange kb = pmb->cellbounds.GetBoundsK(IndexDomain::interior);
```

Comment:
In general I think we want to move to `IndexDomain::interior` for initial conditions, because there's no guarantee that the function providing the initial conditions will be consistent with the chosen boundary conditions. So either way we should be overwriting the ghost cells before the first simulation timestep, but I think since the values from the initial conditions functions in the ghost cells should never be used by the code, it is more clear to use `IndexDomain::interior` in `pgen`.

There was an issue before where boundary conditions weren't always being called before the first timestep so we had to write initial conditions to `IndexDomain::entire`, but again this was inconsistent with our boundaries.